src/v/security/scram_algorithm.h

```diff
@@ -51,6 +52,10 @@ class scram_exception final : public std::exception {
  */
 class client_first_message {
 public:
+    client_first_message(ss::sstring username, ss::sstring nonce)
```

Comment:
unclear to me what these `message()` methods do and how they are used. the commit body doesn't say much.

Reply:
the file is already commented and references https://tools.ietf.org/html/rfc5802. this is clear from context for anyone reading the file.

Reply:
interestingly, `client first message` is literally the name of the message in the protocol spec haha